### PR TITLE
fix: [SPG-786]: Move all secondary node queries to the analytics node

### PR DESCRIPTION
**Pull request by rafaelmart1ns**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/39805)

### CODE REVIEW

**Review by abhinav-harness — DISMISSED** [review]
> Reviewed 23 of 23 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/39805)* status: :shipit: complete! ✓

**Review by Fernando-Dourado — CHANGES_REQUESTED** [review]
> Reviewed 23 of 23 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/39805)* status: :shipit: complete! ✓

___
*[`960-persistence/src/main/java/io/harness/persistence/HPersistence.java` line 170 at r1](https://reviewable.io/reviews/harness/harness-core/39805#-NGO8RsI1toBExrgPc_M:-NGO8RsI1toBExrgPc_N:b-gxk767) ([raw file](https://github.com/harness/harness-core/blob/6d1caf460f423f9a1346cda6c272fc83f90ae627/960-persistence/src/main/java/io/harness/persistence/HPersistence.java#L170)):*
> ```Java
>      * @return the query
>      */
>   <T extends PersistentEntity> Query<T> createAnalyticsQuery(Class<T> cls);
> ```

It is not missing the implementation?

**Review by Fernando-Dourado — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/39805)* status: :shipit: complete! ✓

**Review by Fernando-Dourado — APPROVED** [review]
> Reviewed 20 of 20 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/39805)* status: :shipit: complete! ✓

File: 400-rest/src/main/java/io/harness/execution/export/background/ExportExecutionsService.java [hunk 1]
```diff
@@ -306,7 +306,7 @@ private String prepareFolderName(ExecutionMetadata executionMetadata) {
   @VisibleForTesting
   List<WorkflowExecution> processExportExecutionsRequestBatch(
       @NotNull ExportExecutionsRequest request, int pageOffset) {
-    Query<WorkflowExecution> query = wingsPersistence.createQuery(WorkflowExecution.class, excludeAuthority);
```
Comment (lasdox) [inline 400-rest/src/main/java/io/harness/execution/export/background/ExportExecutionsService.java:309]: need to use the exclude authority here

File: 400-rest/src/main/java/software/wings/service/impl/ActivityServiceImpl.java [hunk 2]
```diff
@@ -129,7 +129,7 @@ public Map<String, List<CommandUnitDetails>> getCommandUnitsMapUsingSecondary(Co
     // Activity doesn't have accountId in it's collection. So excluding authority for now. Use case is to get command
     // units for all activities for a set of executions in an account (may not be the same app). The account check is
     // done on workflow executions.
-    List<Activity> activities = wingsPersistence.createQuery(Activity.class, excludeAuthority)
```
Comment (lasdox) [inline 400-rest/src/main/java/software/wings/service/impl/ActivityServiceImpl.java:132]: need to use the exclude authority here

Comment (rafaelmart1ns) [reply 400-rest/src/main/java/software/wings/service/impl/ActivityServiceImpl.java:132]: done.

File: 400-rest/src/main/java/io/harness/execution/export/background/ExportExecutionsService.java (hunk 1 above)
Comment (rafaelmart1ns) [reply 400-rest/src/main/java/io/harness/execution/export/background/ExportExecutionsService.java:309]: done.

**Review by lasdox — APPROVED** [review]
> Reviewed 2 of 23 files at r1, 20 of 20 files at r2, 1 of 2 files at r3, 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/39805)* status: :shipit: complete! ✓

**Pull request merged**